src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -21,28 +26,64 @@ export interface ICoExpressionTabProps {
+    @observable _selectedQueryProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedGeneticEntityId:string | undefined; // only undefined initially, until genes and gene sets downloaded
+    @observable _geneticEntities:GeneticEntity[] | undefined; // only undefined initially, until genes and gene sets downloaded
+    @observable _isSelectedGeneticEntityAGeneSet = false;
 
-    @computed get selectedMolecularProfile():MolecularProfile|undefined {
-        if (!this._selectedMolecularProfile && this.props.store.coexpressionTabMolecularProfiles.isComplete
-            && this.props.store.coexpressionTabMolecularProfiles.result.length > 0) {
-            return this.props.store.coexpressionTabMolecularProfiles.result[0];
+    @computed get selectedSubjectProfileGene():MolecularProfile|undefined {
+        if (!this._selectedSubjectProfileGene && this.props.store.coexpressionTabMolecularSubjectProfilesGene.isComplete
+                && this.props.store.coexpressionTabMolecularSubjectProfilesGene.result.length > 0) {
+                    return this.props.store.coexpressionTabMolecularSubjectProfilesGene.result[0];
         } else {
-            return this._selectedMolecularProfile;
+            return this._selectedSubjectProfileGene;
         }
     }
 
-    @computed get selectedEntrezGeneId():number | undefined {
-        if (this._selectedEntrezGeneId === undefined && this.props.store.genes.isComplete &&
-                this.props.store.genes.result.length > 0) {
-            return this.props.store.genes.result[0].entrezGeneId;
+    @computed get selectedSubjectProfileGeneset():MolecularProfile|undefined {
+        if (!this._selectedSubjectProfileGeneset && this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.isComplete
+                && this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.result.length > 0) {
+                    return this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.result[0];
         } else {
-            return this._selectedEntrezGeneId;
+            return this._selectedSubjectProfileGeneset;
+        }
+    }
+
+    @computed get selectedQueryProfile():MolecularProfile|undefined {
+        if (!this._selectedQueryProfile && this.props.store.coexpressionTabMolecularQueryProfiles.isComplete
+            && this.props.store.coexpressionTabMolecularQueryProfiles.result.length > 0) {
+            return this.props.store.coexpressionTabMolecularQueryProfiles.result[0];
+        } else {
+            return this._selectedQueryProfile;
+        }
+    }
+
+    @computed get selectedGeneticEntityId():string | undefined {
+        if (!this._selectedGeneticEntityId && this.props.store.geneticEntities !== undefined) {
+            return String(this.props.store.geneticEntities[0].geneticEntityId);
```

Comment:
and please change all instances of String() to .toString() and Number() to parseFloat()

Reply:
I've changed all `String()` to `.toString()` and the `Number()` that were refering to a string to `parseFloat()`.